src/passes/RemoveUnusedModuleElements.cpp

```diff
@@ -712,6 +714,65 @@ struct RemoveUnusedModuleElements : public Pass {
     //       to a function from an element segment, we may be able to remove
     //       that function, etc.)
   }
+
+  // Do simple work that prepares the module to be efficiently optimized.
+  void prepare(Module* module) {
+    // If a function export is a function that just calls another function, we
+    // can export that one directly. Doing so might make the function in the
+    // middle unused:
+    //
+    //  (export "export" (func $middle))
+    //  (func $middle
+    //    (call $real)
+    //  )
+    //
+    // =>
+    //
+    //  (export "export" (func $real))  ;; this changed
+    //  (func $middle
+    //    (call $real)
+    //  )
+    //
+    // (Normally this is not needed, as inlining will end up removing such
+    // silly trampoline functions, but the case of an import being exported does
+    // not have any code for inlining to work with, so we need to handle it
+    // directly.)
+    for (auto& exp : module->exports) {
+      if (exp->kind != ExternalKind::Function) {
+        continue;
+      }
+
+      auto* func = module->getFunction(exp->value);
+      if (!func->body) {
+        continue;
+      }
+
+      auto* call = func->body->dynCast<Call>();
```

Comment:
Should this also check for `ReturnCall`? (assuming that tail calls are represented via different node in Binaryen)

Reply:
ReturnCall is the same as Call, just with `isReturn` set to true, so it's already handled. I'll add a test though for coverage.

Reply:
Ah good to know.